central/sensor/service/pipeline/imageintegrations/pipeline.go

```diff
@@ -54,14 +61,42 @@ func NewPipeline(integrationManager enrichment.Manager,
 type pipelineImpl struct {
 	integrationManager enrichment.Manager
 
+	connManager         connection.Manager
 	datastore           datastore.DataStore
 	clusterDatastore    clusterDatastore.DataStore
 	enrichAndDetectLoop reprocessor.Loop
 }
 
-func (s *pipelineImpl) Reconcile(_ context.Context, _ string, _ *reconciliation.StoreMap) error {
-	// Nothing to reconcile for image integrations
-	return nil
+func (s *pipelineImpl) Reconcile(ctx context.Context, clusterID string, storeMap *reconciliation.StoreMap) error {
+	existingIDs := set.NewStringSet()
+
+	conn := s.connManager.GetConnection(clusterID)
+	if conn == nil {
+		return nil
+	}
+	// We should not reconcile image integrations unless the Sensor is on a version that can provide scoped integrations
+	// with consistent uuids
+	if !conn.HasCapability(centralsensor.ScopedImageIntegrations) {
+		return nil
+	}
+
+	integrations, err := s.datastore.GetImageIntegrations(ctx, &v1.GetImageIntegrationsRequest{})
+	if err != nil {
+		return errors.Wrap(err, "getting image integrations for reconciliation")
+	}
+	for _, integration := range integrations {
+		if integration.GetEcr() != nil {
+			continue
+		}
+		if integration.GetClusterId() != clusterID {
```

Comment:
I looked at `v1.GetImageIntegrationsRequest` and it seems that it supports specifying a cluster (name only, not ID?). However, the handling of that field is extremely weird (`central/imageintegration/datastore/datastore_impl.go:55-64):
```
	integrations, err := ds.storage.GetAll(ctx)
	if err != nil {
		return nil, err
	}

	integrationSlice := integrations[:0]
	for _, integration := range integrations {
		if request.GetCluster() != "" {
			continue
		}
```
So if `request` specifies a non-empty cluster, it will always return an empty slice, while still doing a read on the DB. Not sure if there's a bug or this param was just never supported, but either way, this probably deserves a JIRA filed for it and addressed some time.

Reply:
So, seems like the original PR #658 was actually removing support for this parameter (but apparently, this wasn't done because there was still _something_ left for autogenerated image integrations that uses this parameter).

I do agree, it does justify a JIRA, I created one: [ROX-13071](https://issues.redhat.com/browse/ROX-13071)